src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -183,8 +183,8 @@ Map<String, String> buildMapWithKnnKeyAndOriginalValue(final IngestDocument inge
             if (!sourceAndMetadataMap.containsKey(originalKey)) {
                 continue;
             }
-            if (!(sourceAndMetadataMap.get(originalKey) instanceof String)) {
-                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a string");
+            if (!(sourceAndMetadataMap.get(originalKey) instanceof String) || ((String) sourceAndMetadataMap.get(originalKey)).isEmpty()) {
+                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a non-empty string");
```

Comment:
better have the log entry for the actual value and its class, it can help in troubleshooting if this happens in prod

Reply:
Also please use String.format for the exception message.